packages/grid/_modules_/grid/hooks/features/export/useGridCsvExport.tsx

```diff
@@ -2,34 +2,48 @@ import * as React from 'react';
 import { GridApiRef } from '../../../models/api/gridApiRef';
 import { useGridApiMethod } from '../../root/useGridApiMethod';
 import { useGridSelector } from '../core/useGridSelector';
-import { visibleGridColumnsSelector } from '../columns';
+import { allGridColumnsSelector, visibleGridColumnsSelector } from '../columns';
 import { visibleSortedGridRowsSelector } from '../filter';
 import { gridSelectionStateSelector } from '../selection';
 import { GridCsvExportApi } from '../../../models/api/gridCsvExportApi';
 import { GridExportCsvOptions } from '../../../models/gridExport';
 import { useLogger } from '../../utils/useLogger';
 import { exportAs } from '../../../utils';
 import { buildCSV } from './serializers/csvSerializer';
+import { GridColDef } from '../../../models';
 
 export const useGridCsvExport = (apiRef: GridApiRef): void => {
   const logger = useLogger('useGridCsvExport');
   const visibleColumns = useGridSelector(apiRef, visibleGridColumnsSelector);
+  const columns = useGridSelector(apiRef, allGridColumnsSelector);
   const visibleSortedRows = useGridSelector(apiRef, visibleSortedGridRowsSelector);
   const selection = useGridSelector(apiRef, gridSelectionStateSelector);
 
   const getDataAsCsv = React.useCallback(
     (options?: GridExportCsvOptions): string => {
       logger.debug(`Get data as CSV`);
 
+      let exportedColumns: GridColDef[];
+
+      if (options?.fields) {
+        exportedColumns = options.fields
+          .map((field) => columns.find((column) => column.field === field))
+          .filter((column): column is GridColDef => !!column);
```

Comment:
Oh nice, first time I see this TS feature used: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates, I have learned something 👍

Reply:
I love this feature
It's not usefull very often, but for non-nullish filters it makes the result a lot more readable.